proto/provenance/trigger/v1/event.proto

```diff
@@ -8,12 +8,28 @@ option java_multiple_files = true;
 
 // EventTriggerCreated is an event for when a trigger is created
 message EventTriggerCreated {
-  // trigger_id is a unique identifier of the trigger
+  // trigger_id is a unique identifier of the trigger.
   string trigger_id = 1;
 }
 
 // EventTriggerDestroyed is an event for when a trigger is destroyed
 message EventTriggerDestroyed {
-  // trigger_id is a unique identifier of the trigger
+  // trigger_id is a unique identifier of the trigger.
   string trigger_id = 1;
+}
+
+// EventTriggerDetected is an event for when a trigger's event is detected
+message EventTriggerDetected {
+  // trigger_id is a unique identifier of the trigger.
+  string trigger_id = 1;
+}
+
+// EventTriggerExecuted is an event for when a trigger is executed.
+message EventTriggerExecuted {
+  // trigger_id is a unique identifier of the trigger.
+  string trigger_id = 1;
+  // owner is the creator of the trigger.
+  string owner = 2;
+  // error contains the failure message for an unsuccessful action.
+  string error = 3;
```

Comment:
This isn't a good idea because the output of an error can be non-deterministic depending on the platform.

Reply:
Agreed, a success flag would be safer.